flink-runtime/src/main/java/org/apache/flink/runtime/security/token/DelegationTokenManager.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.security.token;
+
+import org.apache.flink.configuration.Configuration;
+
+import org.apache.hadoop.security.Credentials;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Manager for delegation tokens in a Flink cluster.
+ *
+ * <p>When delegation token renewal is enabled, this manager will make sure long-running apps can
+ * run without interruption while accessing secured services. It periodically logs in to the KDC
+ * with user-provided credentials, and contacts all the configured secure services to obtain
+ * delegation tokens to be distributed to the rest of the application.
+ */
+public class DelegationTokenManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DelegationTokenManager.class);
+
+    private final Configuration configuration;
+
+    final Map<String, DelegationTokenProvider> delegationTokenProviders;
+
+    public DelegationTokenManager(Configuration configuration) {
+        this.configuration = checkNotNull(configuration, "Flink configuration must not be null");
```

Comment:
This is super cool but nothing is enforcing this contract, right? Is this an agreement not to enforce such things?
I can remove it in a second but I prefer meaningful messages instead of NPE.

Reply:
This code is just dropped so the comment is outdated.